app/src/main/java/com/android/bookswap/model/map/BookFilter.kt

```diff
@@ -50,8 +50,9 @@ class BookFilter {
   fun setGenres(genres: List<String>) {
     val newGenresFilter =
         genres.mapNotNull { genre ->
-          // Use the Genre parameter to get the corresponding BookGenres enum
-          BookGenres.values().firstOrNull { it.Genre.equals(genre, ignoreCase = true) }
+          BookGenres.values().firstOrNull {
+            it.Genre.replace("-", "").equals(genre.replace("-", ""), ignoreCase = true)
```

Comment:
This is not where the problem happens, if the genre `SCHOOLMANUAL("School Manual")` is added, the bug will happen again.
The problem stems from the lines §123-124 of the `FilterMap.kt` file where filters are set again but using the enum class members names instead of their `Genre` value.

Reply:
Thank you for your help. My PR fixed the issue with the dash, but I didn't realize that the bug could also occur with spaces. My next commit should resolve this issue.